run.py

```diff
@@ -398,25 +406,28 @@ def run_assembly(asm_test, iss_yaml, isa, mabi, iss_opts, output_dir, setting_di
   if len(iss_list) == 2:
     compare_iss_log(iss_list, log_list, report)
 
-def run_assembly_from_dir(asm_test_dir, iss_yaml, isa, mabi, iss, output_dir, setting_dir):
+def run_assembly_from_dir(asm_test_dir, iss_yaml, isa, mabi, gcc_opts, iss,
+                          output_dir, setting_dir):
   """Run a directed assembly test from a directory with spike
 
   Args:
     asm_test_dir    : Assembly test file directory
     iss_yaml        : ISS configuration file in YAML format
     isa             : ISA variant passed to the ISS
     mabi            : MABI variant passed to GCC
+    gcc_opts        : User-defined options for GCC compilation
     iss             : Instruction set simulators
     output_dir      : Output directory of compiled test files
     setting_dir     : Generator setting directory
   """
-  result = run_cmd("find %s -name \"*.S\"" % asm_test_dir)
+  result = run_cmd("find %s -maxdepth 1 -name \"*.S\"" % asm_test_dir)
```

Comment:
why adding this constraint?

Reply:
without this constraint here, the flow recursively searches every subdirectory for assembly files, which may not be intended behavior

Reply:
I think it's better to have the ability to search through multiple files.